client/web-sveltekit/src/lib/repo/RepoPopover/RepoPopover.svelte

```diff
@@ -1,184 +1,152 @@
-                </div>
+            <div class="left">
+                <Icon svgPath={mdiSourceMerge} --icon-fill-color="var(--primary)" />
+                <h4>{displayRepoName(data.name)}</h4>
+                <Badge variant="outlineSecondary" small pill>
+                    {data.isPrivate ? 'Private' : 'Public'}
+                </Badge>
             </div>
-            <div class="code-host">
-                <Icon svgPath={codeHostIcon} --color="var(--text-body)" --icon-size="24px" />
-                <div><small>{capitalize(codeHostKind)}</small></div>
+            <div class="right">
+                <Icon svgPath={getIconPathForCodeHost(codeHostKind)} --icon-fill-color="var(--text-body)" --size={24} />
+                <small>{capitalize(codeHostKind)}</small>
             </div>
         </div>
-        <div class="divider" />
     {/if}
 
-    <div class="description-and-tags">
-        <div class="description">{repo.description}</div>
-        <div class="tags">
-            {#if repo.tags.nodes.length > 0}
-                {#each repo.tags.nodes as tag}
-                    <div class="tag"><small>{tag.name}</small></div>
-                {/each}
+    {#if data.description || data.topics.length}
+        <div class="description-and-tags">
+            <div class="description">
+                {data.description}
+            </div>
+            {#if data.topics.length}
+                <div class="tags">
+                    {#each data.topics as topic}
+                        <Badge variant="link" small pill>{topic}</Badge>
+                    {/each}
+                </div>
             {/if}
         </div>
-    </div>
-
-    <div class="divider" />
+    {/if}
 
-    <div class="last-commit">
-        <div class="heading">
+    {#if commit}
+        <div class="last-commit">
             <small>Last Commit</small>
-        </div>
 
-        <div class="commit-info">
-            <div class="commit">
-                <!--
-                A <div> element is needed for subject and commit message
-                because the <small> element alone doesn't work with
-                text-overflow: ellipsis.
-                -->
-                <div class="subject">
-                    <small>{subject}</small>
-                </div>
-                {#if commitSHA}
-                    <div class="commit-number">
-                        <small class="commit-number"
-                            ><a href={url} target="_blank">#{truncateCommitNumber(commitSHA, 6)}</a></small
-                        >
+            <div class="commit-info">
+                <small class="subject"><a href={commit.canonicalURL}>{commit.subject}</a></small>
+                {#if author && avatar}
+                    <div class="author">
+                        <Avatar {avatar} --avatar-size="1.0rem" />
+                        <small>{avatar.displayName} · <Timestamp date={author?.date} /></small>
```

Comment:
I think this should work (i.e. the `author?.avatar` check will narrow down the type to `author` and `avatar` being defined). I don't see any advantage of extracting `avatar` here.
```suggestion
                {#if author?.avatar}
                    <div class="author">
                        <Avatar avatar={author.avatar} --avatar-size="1.0rem" />
                        <small>{author.avatar.displayName} · <Timestamp date={author.date} /></small>
```